api/_db.data-utils.js

```diff
@@ -105,28 +139,32 @@ export const findById = (data, id) => {
 		return row[filterColumnIdx] === id;
 	});
 
-	// no result
-	assert(filteredRows.length, 'Not found');
+	const parsedCertificates = filteredRows.map((row) => {
+		const certificate = normalizeCertificate(row[certificateColumnIdx] ?? '');
 
-	// more than one result, assume an filtering error
-	if (filteredRows.length > 1) {
-		console.log('Filtered more than one row', filteredRows);
-		throw new Error('More than one row');
-	}
+		const formFillDate = parseDate(row[formFillTimeColumnIdx]);
+		const expiryDate = parseDate(row[expiryDateColumnIdx]) || getExpiryTimeFromFormFillTime(formFillDate) || null;
+
+		return {
+			id,
+			certificate,
+			name: row[nameColumnIdx],
+			examiner: row[examinerColumnIdx] || null,
+			examTime: formatDate(parseDate(row[examDateColumnIdx])),
```

Comment:
Let's parse(at the time of loading the info, eg here) and format(at the very end when returning from here) each of the dates once.

Reply:
Good idea, this makes everything easier!